### PR TITLE
docs: update README for v4

### PR DESCRIPTION
**Pull request by Belco90**
Relates to #186 

I'm not sure if the Shared Settings description should be the same on README and migrating guide 🤔 . What do you think? /cc @MichaelDeBoey @CreativeTechGuy

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,6 +146,107 @@ To enable this configuration use the `extends` property in your
 }
 ```
 
+## Configuration
+
+There are some configuration options available that will be shared across all the plugin rules. This is achieved using [ESLint Shared Settings](https://eslint.org/docs/user-guide/configuring/configuration-files#adding-shared-settings). These shared settings are meant to be used if you need to restrict the Aggressive Reporting mechanism, **so please before configuring any of these settings**, read more about [`eslint-plugin-testing-library` Aggressive Reporting mechanism](docs/migrating-to-v4-guide.md#aggressive-reporting), and [how it's affected by these settings](docs/migrating-to-v4-guide.md#shared-settings).
```
Comment (CreativeTechGuy) [inline README.md:151]: I think it'd be valuable to call out here that restricting Aggressive Reporting is a specialized/advanced feature. The name makes it seem bad and scary and might make people think they need to configure this. In reality, this is not needed for most people to ever touch.

File: README.md [hunk 2]
```diff
@@ -123,6 +146,107 @@ To enable this configuration use the `extends` property in your
...
+
+If you are sure about configuring the settings, these are the options available:
```
Comment (CreativeTechGuy) [inline README.md:153]: Does it make sense to put all of this in the main README given that it's not something that most users would need? Maybe a separate page that explains this in detail and this document can just contain the summary and explanation of what can be configured and why someone might need to configure it. Probably doesn't need to go into such detail about how to actually do it.

Comment (Belco90) [reply README.md:153]: Yeah, I think I'm gonna move this Configuration section to the very bottom of the README, put this detailed explanation into migrating guide, and leave just the "how to" here, referencing the detailed version.

File: README.md [hunk 3]
```diff
@@ -54,22 +62,37 @@ Add `testing-library` to the plugins section of your `.eslintrc` configuration f
...
+// ❌ Don't do this
+{
+  "extends": ["plugin:testing-library/dom", "plugin:testing-library/react"]
+}
+
+// ✅ Do just this instead
+{
+  "extends": ["plugin:testing-library/react"]
+}
+```
```
Comment (gndelia) [inline README.md:95]: nice example here, as it might confuse people 👏

**Review by gndelia — APPROVED** [review]
Looks great 🚀  I love how it is looking!

**Pull request merged**